File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int j = 0;
        
        for (int i = 0; i < n; i++) {
            if (i == 0 || i == 1 || nums[i] != nums[j - 2])
                nums[j++] = nums[i];
        }
        
        return j;
    }
};

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.ts
function removeDuplicates(nums: number[]): number {
    let j = 0;

    for (let i = 0; i < nums.length; i++) {
        if (i < 2 || nums[i] !== nums[j - 2]) {
            nums[j++] = nums[i];
        }
    }

    return j;
};